netsim/defaults/attributes.yml

```diff
@@ -80,6 +80,7 @@ interface:                  # Interface (node-to-link attachment) attributes
   ipv6: { type: ipv6, use: interface }
   ifindex: int
   ifname: str
+  type: { type: str, valid_values: [ p2p, lan, stub, loopback, vlan_member, svi, lag ] }
```

Comment:
The "type" attribute is set when the interface is created during the link transformation, so it does not need to be defined (the validation is done before that).

It also makes no sense to allow the user to set the interface type in the lab topology, and not having an attribute defined helps to enforce that ;) This needs to be removed.

Reply:
When I leave it out, I get this:
```
Reading topology/input/lag-l2.yml
[ERRORS]  Errors found in lag-l2.yml
[VALUE]   links: attribute links[1].r1.type has invalid value(s): lag
[HINT]    valid values are: lan,p2p,stub,loopback,tunnel,vlan_member,virtual_lag
          use 'netlab show attributes interface' to display valid attributes
[VALUE]   links: attribute links[1].r2.type has invalid value(s): lag
[HINT]    valid values are: lan,p2p,stub,loopback,tunnel,vlan_member,virtual_lag
[FATAL]   Cannot proceed beyond this point due to errors, exiting
```

So either it does need to be defined, or that validation logic needs to be modified

Reply:
The ```lag``` module is currently adding interfaces with type ```lag``` during ```module_pre_transform``` - need to postpone that to ```module_pre_link_transform```

Reply:
...which does not work either, because then the ```vlan``` module must be changed from ```pre_link_transform``` to ```link_pre_link_transform``` (as it assumes link.interfaces is populated) but then the attribute validation stage misses user provided attributes

Reply:
Finally arrived at the following solution:

1. Create lag interfaces during ```module_pre_transform``` such that they are present when ```vlan.link_pre_transform``` executes
2. Use ```_type: lag``` to create them such that validation ignores the attribute, then rename in ```link_pre_link_transform``` (after validation)

It is not possible to create the interfaces after validation, because validation needs to consider user provided interface attributes, some of which depend on vlan transformation